benchmarks/src/main.rs

```diff
@@ -61,8 +49,8 @@ fn get_nth_percentile(durations: &[Duration], n: f64) -> Duration {
     durations[std::cmp::max(index, 1) - 1]
```

Comment:
this function needs to update its usage of `MAX_ITERATIONS` to `durations.len()`. Once this is fixed we can remove the restriction in `score_test` that the various other percentiles are only used when `MAX_ITERATIONS == 100`

Reply:
I'm not sure you're correct about this one; the specification says the following about the formula:

```
For each micro-benchmark, the 10th, 25th, 50th, 75th, 90th, 95th, 98th and 99th percentiles will be recorded using the following algorithm:

* Given a 0-indexed array A of N iteration wall clock times
* Sort the array into ascending order (i.e. shortest time first)
* Let the index i for percentile p in the range [1,100] be defined as: i = int(N * p / 100) - 1
```

In this case, the algorithm's `n` is `durations.len()` in Rust, the algorithm's `p` is (confusingly) `n` in Rust, and `100` is `MAX_ITERATIONS` in Rust (because we allow the user to configure it). If we just used `durations.len()` again, it would be `int(N * p / N) - 1`, which is just `p - 1`. That would work if we always completed 100 measurements, but if we did, we wouldn't need this formula in the first place! The idea here is that if we _don't_ complete 100 measurements, we estimate calculate which index in the sorted list of durations is closest to each percentile; for example, if we only finished 20 runs, then the 50th percentile would be index `(20 * 50 / 100) - 1`, or index `9` in the vector, or the 10th element out of 20.

Reply:
The 100 here is just converting `p` which is in [1,100] to an actual percentage in [0.01, 1], it's just a coincidence that `MAX_ITERATIONS` is also 100. You're right that the `durations.len()` in the numerator needs to be removed too, though. The algorithm here is as simple as `index = array length * fraction - 1`, where fraction is `p/100` and `array length` is `N` or the number of iterations. So if we're looking for the median, fraction is 0.5 and we get the middle element.

As a side note, `MAX_ITERATIONS` is somewhat poorly named, since it isn't necessarily a maximum. It's also not a minimum though. I'm not exactly sure what to call it.

Reply:
Okay, I'll update it. As for a new name, how about `TARGET_ITERATION_COUNT`?

Reply:
Yeah that sounds good :+1:

Reply:
I think the `as usize` needs to come after the division. As it is now, `50.0 as usize / 100` will be 0.

Reply:
Fixed

Reply:
Sorry for the extra round trip, but i think we also need an `as f64` for the multiplication portion, and then the end result is the only thing cast to `usize`.